src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -2,319 +2,263 @@
-            localruntrack[--localruntrackpos] = I2;
+            localruntrack[--localruntrackpos] = i1;
+            localruntrack[--localruntrackpos] = i2;
             localruntrack[--localruntrackpos] = -_codepos;
 
             runtrackpos = localruntrackpos;
         }
 
         private void Backtrack()
         {
-            int newpos = runtrack![runtrackpos++];
+            int newpos = runtrack![runtrackpos];
+            runtrackpos++;
+
 #if DEBUG
             if (runmatch!.IsDebug)
             {
-                if (newpos < 0)
-                    Debug.WriteLine("       Backtracking (back2) to code position " + (-newpos));
-                else
-                    Debug.WriteLine("       Backtracking to code position " + newpos);
+                Debug.WriteLine(newpos < 0 ?
+                    $"       Backtracking (back2) to code position {-newpos}" :
+                    $"       Backtracking to code position {newpos}");
             }
 #endif
 
+            int back = RegexCode.Back;
             if (newpos < 0)
             {
                 newpos = -newpos;
-                SetOperator(_code.Codes[newpos] | RegexCode.Back2);
-            }
-            else
-            {
-                SetOperator(_code.Codes[newpos] | RegexCode.Back);
+                back = RegexCode.Back2;
             }
+            SetOperator(_code.Codes[newpos] | back);
 
-            // When branching backward, ensure storage
+            // When branching backward, ensure storage.
             if (newpos < _codepos)
+            {
                 EnsureStorage();
+            }
 
             _codepos = newpos;
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private void SetOperator(int op)
         {
-            _caseInsensitive = (0 != (op & RegexCode.Ci));
-            _rightToLeft = (0 != (op & RegexCode.Rtl));
             _operator = op & ~(RegexCode.Rtl | RegexCode.Ci);
+            _caseInsensitive = (op & RegexCode.Ci) != 0;
+            _rightToLeft = (op & RegexCode.Rtl) != 0;
         }
 
-        private void TrackPop()
-        {
-            runtrackpos++;
-        }
+        private void TrackPop() => runtrackpos++;
 
-        // pop framesize items from the backtracking stack
-        private void TrackPop(int framesize)
-        {
-            runtrackpos += framesize;
-        }
+        /// <summary>Pop framesize items from the backtracking stack.</summary>
+        private void TrackPop(int framesize) => runtrackpos += framesize;
 
-        // Technically we are actually peeking at items already popped.  So if you want to
-        // get and pop the top item from the stack, you do
-        // TrackPop();
-        // TrackPeek();
-        private int TrackPeek()
-        {
-            return runtrack![runtrackpos - 1];
-        }
+        /// <summary>
+        /// Technically we are actually peeking at items already popped.  So if you want to
```

Comment:
Again, this is sort of a weird summary comment of this method.

Reply:
Fixed.